Model/Quote/Checkout.php

```diff
@@ -160,14 +160,16 @@ private function getDeliveryData(): array
             $signatureFee     = $this->helper->getMethodPrice($carrierPath[$carrier], 'delivery/signature_fee', false);
             $onlyRecipientFee = $this->helper->getMethodPrice($carrierPath[$carrier], 'delivery/only_recipient_fee', false);
             $isAgeCheckActive = $this->isAgeCheckActive($carrierPath[$carrier]);
+            $mailboxPackage   = $this->getPackageType()['packageType'] === 'mailbox';
+            $showPickup       = $this->helper->getBoolConfig($carrierPath[$carrier], 'mailbox/pickup_mailbox');
 
             $myParcelConfig['carrierSettings'][$carrier] = [
                 'allowDeliveryOptions'  => $this->package->deliveryOptionsDisabled ? false : $this->helper->getBoolConfig($carrierPath[$carrier], 'delivery/active'),
                 'allowSignature'        => $this->helper->getBoolConfig($carrierPath[$carrier], 'delivery/signature_active'),
                 'allowOnlyRecipient'    => $this->helper->getBoolConfig($carrierPath[$carrier], 'delivery/only_recipient_active'),
                  'allowMorningDelivery' => $isAgeCheckActive ? false : $this->helper->getBoolConfig($carrierPath[$carrier], 'morning/active'),
-                'allowEveningDelivery' => $isAgeCheckActive ? false : $this->helper->getBoolConfig($carrierPath[$carrier], 'evening/active'),
-                'allowPickupLocations'  => $this->package->deliveryOptionsDisabled ? false : $this->helper->getBoolConfig($carrierPath[$carrier], 'pickup/active'),
+                'allowEveningDelivery'  => $isAgeCheckActive ? false : $this->helper->getBoolConfig($carrierPath[$carrier], 'evening/active'),
+                'allowPickupLocations'  => $this->package->deliveryOptionsDisabled || ($mailboxPackage && !$showPickup) ? false : $this->helper->getBoolConfig($carrierPath[$carrier], 'pickup/active'),
```

Comment:
Misschien is het mooier om de checks die je hier doet in een nieuwe/bestaande method te plaatsen. Dan hoef je hier alleen maar de method aan te roepen.